enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/pipes/UnionSlottedPipe.scala

```diff
@@ -37,82 +35,8 @@ case class UnionSlottedPipe(lhs: Pipe, rhs: Pipe,
-        val in = left.next()
-        val out = PrimitiveExecutionContext(unionInfo)
-        mapSlots.foreach(f => f(in, out, lhsInfo, state))
-        out
-      }
-      else {
-        val in = right.next()
-        val out = PrimitiveExecutionContext(unionInfo)
-        mapSlots.foreach(f => f(in, out, rhsInfo, state))
-        out
-      }
-    }
-  }
-}
-
-case class FastLeftHandUnionSlottedPipe(lhs: Pipe, rhs: Pipe,
-                            rhsInfo: PipelineInformation,
-                            unionInfo: PipelineInformation,
-                            mapSlots: Iterable[(ExecutionContext, ExecutionContext, PipelineInformation, QueryState) => Unit])
-                           (val id: LogicalPlanId = LogicalPlanId.DEFAULT) extends Pipe {
-
-  protected def internalCreateResults(state: QueryState): Iterator[ExecutionContext] = {
-    val left = lhs.createResults(state)
-    val right = rhs.createResults(state)
-
-    new Iterator[ExecutionContext] {
-      override def hasNext: Boolean = left.hasNext || right.hasNext
-
-      override def next(): ExecutionContext = if (left.hasNext) left.next()
-      else {
-        val in = right.next()
-        val out = PrimitiveExecutionContext(unionInfo)
-        mapSlots.foreach(f => f(in, out, rhsInfo, state))
-        out
-      }
-    }
-  }
-}
-
-case class FastRightHandUnionSlottedPipe(lhs: Pipe, rhs: Pipe,
-                            lhsInfo: PipelineInformation,
-                            unionInfo: PipelineInformation,
-                            mapSlots: Iterable[(ExecutionContext, ExecutionContext, PipelineInformation, QueryState) => Unit])
-                           (val id: LogicalPlanId = LogicalPlanId.DEFAULT) extends Pipe {
-
-  protected def internalCreateResults(state: QueryState): Iterator[ExecutionContext] = {
-    val left = lhs.createResults(state)
-    val right = rhs.createResults(state)
-
-    new Iterator[ExecutionContext] {
-      override def hasNext: Boolean = left.hasNext || right.hasNext
-
-      override def next(): ExecutionContext = if (left.hasNext) {
-        val in = left.next()
-        val out = PrimitiveExecutionContext(unionInfo)
-        mapSlots.foreach(f => f(in, out, lhsInfo, state))
-        out
-      }
-      else right.next()
-    }
-  }
-}
-
-case class FastUnionSlottedPipe(lhs: Pipe, rhs: Pipe)
-                               (val id: LogicalPlanId = LogicalPlanId.DEFAULT) extends Pipe {
-  // This pipe is only applicable if the ordering of the variables on the lhs equals the ordering on the rhs
-
-  protected def internalCreateResults(state: QueryState): Iterator[ExecutionContext] = {
-    val left = lhs.createResults(state)
-    val right = rhs.createResults(state)
-    new Iterator[ExecutionContext] {
-      override def hasNext: Boolean = left.hasNext || right.hasNext
-      override def next(): ExecutionContext = if (left.hasNext) left.next()
-      else right.next()
```

Comment:
It seems to me that these specialized "Fast" pipes would be faster in the applicable cases, since it avoids a lot of row copying. It also seems to me to be the most common cases. Did you delete them mainly to simplify the code? Did you benchmark any differences? The same RowMapping optimization could also be applied to these specializations.